### PR TITLE
Miscellaneous

### PR DESCRIPTION
**Pull request by GabrielYik**
* Add in new tests for util methods
* Add in sample test cases in DG appendix
* Add correct some errors in DG

### CODE REVIEW

**Comment by GabrielYik** [comment]
> Now that you are at it, I suggest changing `private static final String ALL_NUMBERS_REGEX = "[0-9]+";` to `[0-9]{1,9}` so that we don't overflow the string -> int conversion.

I think I'll just leave it as it is bah. Also cos the regex was made to check whether a string is all numbers and I already made the function for checking under and over flow haha

File: src/test/java/seedu/planner/commons/util/StringUtilTest.java [hunk 1]
```diff
@@ -159,6 +161,24 @@ public void getDetails_nullGiven_throwsNullPointerException() {
         StringUtil.getDetails(null);
     }
 
+    //------------- Tests for willUnderOrOverflow -----------------------------
+
+    @Test
+    public void willUnderOrOverflow_returnsTrue() {
+        // Underflows
+        assertTrue(willUnderOrOverflow(Long.toString(Integer.MIN_VALUE - 1)));
```
Comment (rongjiecomputer) [inline src/test/java/seedu/planner/commons/util/StringUtilTest.java]: Try to test "111111111111111111111111111111111" too.

File: src/main/java/seedu/planner/commons/util/StringUtil.java [hunk 2]
```diff
@@ -75,11 +75,28 @@ public static boolean isNonZeroUnsignedInteger(String s) {
 
     //@@author GabrielYik
 
+    /**
+     * Checks if the String, which is a number, can be parsed by {@code Integer#parseInt}.
+     *
+     * @param s The String
+     * @return True if the String can be parsed, false otherwise
+     */
+    public static boolean isParsable(String s) {
```
Comment (rongjiecomputer) [inline src/main/java/seedu/planner/commons/util/StringUtil.java]: `isParsableInteger`?

File: src/test/java/seedu/planner/commons/util/StringUtilTest.java (hunk 1 above)
Comment (GabrielYik) [reply src/test/java/seedu/planner/commons/util/StringUtilTest.java]: Ok

File: src/main/java/seedu/planner/commons/util/StringUtil.java (hunk 2 above)
Comment (GabrielYik) [reply src/main/java/seedu/planner/commons/util/StringUtil.java]: Ok

**Pull request merged**